src/main/java/eu/fasten/vulnerabilityproducer/utils/parsers/ParserManager.java

```diff
@@ -82,55 +94,105 @@ public ParserManager(JavaHttpClient client,
+    public void getVulnerabilitiesFromParsers(boolean updatesOnly, int numThreadsToUse) {
         if(updatesOnly) {
             logger.info("Updating Versions of Packages");
             versionRanger.updateMappings();
         }
 
-        logger.info("Gathering vulnerabilities from GitHub Advisories");
-        var mapFromGH = ghParser.getVulnerabilities(updatesOnly);
-        logger.info("Gathering vulnerabilities from Extra Sources");
-        var mapFromExtraSources = extraParser.getVulnerabilities(updatesOnly);
-        logger.info("Gathering vulnerabilities from NVD");
-        var mapFromNVD = nvdParser.getVulnerabilities(updatesOnly);
-        logger.info("Gathering vulnerabilities from Debian OVAL");
-        var mapFromOVAL = ovalParser.getVulnerabilities(updatesOnly);
-        logger.info("Gathering vulnerabilities from oss-fuzz-vulns");
-        var mapFromOSSFuzz = ossFuzzParser.getVulnerabilities(updatesOnly);
-
-        logger.info("Merging all the pulled vulnerabilities");
-        var maps = new ArrayList<HashMap<String, Vulnerability>>();
-        maps.add(mapFromNVD);
-        maps.add(mapFromExtraSources);
-        maps.add(mapFromGH);
-        maps.add(mapFromOVAL);
-        maps.add(mapFromOSSFuzz);
-        var vulnerabilities = mergeMapsOfVulnerabilities(maps);
-
-        logger.info("Injecting information about Patches");
-
-        vulnerabilities.forEach(v -> {
+        final var vulnerabilities = parseVulnerabilities(updatesOnly);
+        logger.info("Filtering out vulnerabilities without a forge and storing them to the disk");
+        final var smtQueue = new ConcurrentLinkedQueue<>(vulnerabilities.stream().filter((v) -> !vulnerabilityForgeIsSupported(v)).collect(Collectors.toList()));
+
+        // For writing vuln. statements without blocking other threads
+        Thread smtWriter = new Thread(() -> smtQueue.forEach(this::storeStatement));
+        smtWriter.start();
+
+        logger.info("Injecting information about Patches  using " + numThreadsToUse  + " threads");
+        var vulnerabilitiesWithForge = vulnerabilities.stream().filter(this::vulnerabilityForgeIsSupported).collect(Collectors.toList());
+
+        var executor = Executors.newFixedThreadPool(numThreadsToUse);
+        List<Callable<Object>> patchFindingTasks = new ArrayList<>();
+        try(ProgressBar pb = new ProgressBar("InjectPatches", vulnerabilitiesWithForge.size())) {
+            for (var v: vulnerabilitiesWithForge) {
+                patchFindingTasks.add(() -> {parsePatchesAndPublish(smtQueue, v, pb);
+                    return null;
+                });
+            }
             try {
-                var patchFinder = patchFinderFactory.getNewPatchFinder();
-                logger.info("Inferring purls for " + v.getId());
-                purlMapper.inferPurls(v, patchFinder);
-                if(vulnerabilityForgeIsSupported(v)) {
```

Comment:
Removing this check will cause CVEs for unsupported ecosystems to be fully parsed. That does slow things down. What's the benefit?

Reply:
The slowdown is not problematic however, let's keep it in mind, but proceed with this PR.